monai/networks/blocks/crossattention.py

```diff
@@ -145,23 +165,34 @@ def forward(self, x: torch.Tensor, context: Optional[torch.Tensor] = None):
         q = q.view(b, t, self.num_heads, c // self.num_heads).transpose(1, 2)  # (b, nh, t,  hs)
         k = k.view(b, kv_t, self.num_heads, c // self.num_heads).transpose(1, 2)  # (b, nh, kv_t, hs)
         v = v.view(b, kv_t, self.num_heads, c // self.num_heads).transpose(1, 2)  # (b, nh, kv_t, hs)
-        att_mat = torch.einsum("blxd,blyd->blxy", q, k) * self.scale
 
-        # apply relative positional embedding if defined
-        att_mat = self.rel_positional_embedding(x, att_mat, q) if self.rel_positional_embedding is not None else att_mat
+        if self.use_flash_attention:
+            x = torch.nn.functional.scaled_dot_product_attention(
+                query=q.transpose(1, 2),
```

Comment:
Perhaps you can directly remove the transpose in Line165-Line167. I didn't test cross attention. Could you please test whether the  shape change here is right.

Reply:
The cases for cross attention and self attention are not fully equal.

Reply:
We're working on getting the cross and self attention modules to replicate the results from GenerativeModels so slightly suboptimal code like this will be here. I'd suggest we finish up this PR, integrate the other PRs relating to these classes, verify it performs, and then work on cleaning up these sorts of things.

Reply:
Will address in https://github.com/Project-MONAI/MONAI/pull/7996